alphabet/alphabet.go

```diff
@@ -293,24 +294,21 @@ func NewPairing(s, c string) (*Pairing, error) {
 			return nil, errors.New("alphabet: pairing definition is not a bijection")
 		}
 	}
-
+	copy(p.complements[:], p.pair)
+	for i, ok := range p.ok {
+		if !ok {
+			p.complements[i] |= unicode.MaxASCII + 1
+		}
+	}
 	return p, nil
 }
 
 // Returns the complement of a letter and true if the complement is a valid letter otherwise unchanged and false.
 func (p *Pairing) Complement(l Letter) (c Letter, ok bool) { return p.pair[l], p.ok[l] }
 
 // Returns a complementation table based on the internal representation. Invalid pairs hold a value outside the ASCII range.
-func (p *Pairing) ComplementTable() (t []Letter) {
-	t = make([]Letter, 256)
-	copy(t, p.pair)
-	for i, ok := range p.ok {
-		if !ok {
-			t[i] |= unicode.MaxASCII + 1
-		}
-	}
-
-	return
+func (p *Pairing) ComplementTable() []Letter {
```

Comment:
I think it's worth adding that the returned value should not be altered (this was probably why I wrote it the way I did).